addons/phone_validation/models/models.py

```diff
@@ -20,6 +20,25 @@ def _phone_get_number_fields(self):
             number_fname for number_fname in ('mobile', 'phone') if number_fname in self
         ]
 
+    def _phone_get_country(self):
```

Comment:
In mail API is generally returning a record-based dict( aka record.id: found stuff), allowing to do operations in batch. It is a bit more noisy (in single record you have to do self._phone_get_country()[self.id]) but I think it is better to be batch-enabled.